src/api/users/src/routes/users.js

```diff
@@ -28,27 +28,41 @@ router.get('/:id', validateId(), async (req, res, next) => {
 });
 
 // get all users
-// rejected if the user collection is empty, otherwise users returned
-router.get('/', async (req, res, next) => {
-  try {
-    const usersRef = db.collection('users');
-    const snapshot = await usersRef.get();
+// rejected if the users collection is empty
+// otherwise n (specified via params) users are returned
+router.get('/', validateGetParams(), async (req, res, next) => {
+  // if req.query.page is 0 or unparsable set to 1, otherwise set to the req.query.page
+  const parseReqPage = parseInt(req.query.page, 10);
+  const page = !parseReqPage ? 1 : parseReqPage;
+  // if req.query.per_page is 0 or unparsable, set to 20, otherwise set to the min of either req.query.page or 20
+  const parseReqPerPage = Math.min(parseInt(req.query.per_page, 10), 20);
+  const perPage = !parseReqPerPage ? 20 : parseReqPerPage;
```

Comment:
Same as above comment

Reply:
However entering a negative number here breaks stuff. What do you think about wrapping it in Math.abs to force the number to be positive?

`const parsePerPage = Math.abs(Math.min(parseInt(req.query.per_page, 10), 20));`

Reply:
Actually that might be a little hacky. What about this?

`const parsePerPage = req.query.per_page < 0 ? 20 : Math.min(parseInt(req.query.per_page, 10), 20);` (20 is what we're using as our default)

Reply:
I'm fine with that suggestion, although I'd break it up since `parsing` a value is different than checking perPage. I'd keep line 38 as-is. Then change line 39 to do the conditional check with an or statement.

Also, I think @manekenpix might hate magic numbers (for default value) though and prefers it being set in an env.

Reply:
Okay awesome, I just hardcoded both page and perPage to disallow numbers less than 1.

I don't think setting this in an env is necessary, though I do appreciate it being a magic number. However, I can always change it if requested. It'll just get a bit hairy as I'll have to require in the env file in case we're running this outside of docker mode (i.e. `npm run dev`), which AFAIK we're trying to move away from.